### DISCUSSION TITLE
Adding more Integration and e2e tests

### DISCUSSION THREAD

**Discussion by Sonichigo** [authelia/authelia#6248] [Ideas]
I was going through the codebase and pr, to see how ways to contribute to the authelia. I noticed that their are few tests written but couldn't see more integrations and e2e. I use a tool called Keploy to for testing and generating datamocks, I think it would be a nice to have some more-testcases. Can i contirbute in this way?

**Comment by james-d-elliott** [comment]
If you would like to add to our existing integration test suite we'd welcome it. You can probably add keploy to a new suite if you felt it'd add to coverage. I'd probably advise to check the v4.38.0-beta3 branch as we're about to make some measurable changes there.

**Reply by Sonichigo** [reply]
I'll add to keploy to a new suite then!! Thanks for fast response @james-d-elliott ! 😄

**Reply by james-d-elliott** [reply]
Check out my ninja edit.
